GitCommands/BatchArgumentItem.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace GitCommands
+{
+    /// <summary>
+    /// Result model for batch processing arguments and count of items for batch progress
+    /// </summary>
+    public sealed class BatchArgumentItem
+    {
+        public BatchArgumentItem(ArgumentString argument, int count)
+        {
+            Argument = argument;
+            Count = count;
+        }
+
+        /// <summary>
+        /// Batch command line argument
+        /// </summary>
+        public ArgumentString Argument { get; }
+
+        /// <summary>
+        /// Count of items in batch, used for batch progress update
+        /// </summary>
+        public int Count { get; }
```

Comment:
A `Count` property is self explanatory for a collection type, however here it isn't as self descriptive.
I propose to rename to make it clearer to consumer what it is:
```suggestion
        public int BatchItemsCount { get; }
```